src/NHibernate.Test/NHSpecificTest/NH3848/CriteriaTestFixture.cs

```diff
@@ -0,0 +1,140 @@
+﻿using System.Collections.Generic;
+using System.Threading.Tasks;
+using NHibernate.Criterion;
+using NHibernate.SqlCommand;
+using NHibernate.Transform;
+
+namespace NHibernate.Test.NHSpecificTest.NH3848
+{
+    public class CriteriaTestFixture : TestFixture
+    {
+        protected override IList<Customer> GetCustomersWithFetchedOrdersWithoutRestrictions(ISession session)
+        {
+            return session.CreateCriteria<Customer>()
+                          .CreateAlias("Orders", "order", JoinType.LeftOuterJoin)
+                          .SetResultTransformer(new DistinctRootEntityResultTransformer())
+                          .List<Customer>();
+        }
+
+        protected override IList<Customer> GetCustomersWithOrdersEagerLoaded(ISession session)
+        {
+            return session.CreateCriteria<Customer>()
+					      .Fetch("Orders")
+						  .SetResultTransformer(new DistinctRootEntityResultTransformer())
+                          .List<Customer>();
+        }
+
+		protected override async Task<IList<Customer>> GetCustomersWithOrdersEagerLoadedAsync(ISession session)
```

Comment:
In most cases, no async code should be written. That is the job of the async generator, included for tests. See [contributing](https://github.com/nhibernate/nhibernate-core/blob/master/CONTRIBUTING.md). Is there any special reason here for hand coding them?

By the way, indentation is mixed between spaces and tabs in this file. It should be re-formated to tabs, as instructed in [contributing](https://github.com/nhibernate/nhibernate-core/blob/master/CONTRIBUTING.md#submit-a-pull-request).

(And other .cs files fully spaces indented added by this PR should be switched to tabs.)